opencdc/fromproto.go

```diff
@@ -0,0 +1,94 @@
+// Copyright © 2023 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencdc
+
+import (
+	"fmt"
+
+	opencdcv1 "github.com/conduitio/conduit-commons/proto/opencdc/v1"
+)
+
+func _() {
+	// An "invalid array index" compiler error signifies that the constant values have changed.
+	var cTypes [1]struct{}
+	_ = cTypes[int(OperationCreate)-int(opencdcv1.Operation_OPERATION_CREATE)]
+	_ = cTypes[int(OperationUpdate)-int(opencdcv1.Operation_OPERATION_UPDATE)]
+	_ = cTypes[int(OperationDelete)-int(opencdcv1.Operation_OPERATION_DELETE)]
+	_ = cTypes[int(OperationSnapshot)-int(opencdcv1.Operation_OPERATION_SNAPSHOT)]
+}
+
+func (r *Record) FromProto(proto *opencdcv1.Record) error {
+	if proto == nil {
+		*r = Record{}
```

Comment:
This could be racy.

Reply:
Don't share your records with other goroutines, if you do, don't keep a reference to the record. Also, use `Record` as a value. That's the approach we use in Conduit, so I think we should be good there.

Reply:
I think the above makes sense, but we should have a word or two somewhere to highlight this philosophy in Conduit.